app/src/main/java/com/example/mystore/networking/API.kt

```diff
@@ -1,8 +1,21 @@
 package com.example.mystore.networking
 
-import com.example.mystore.models.ProductModel
+import com.example.mystore.model.LoginModel
+import com.example.mystore.model.LoginResponse
+import com.example.mystore.model.ProductModel
+import retrofit2.Response
+import retrofit2.http.Body
 import retrofit2.http.GET
+import retrofit2.http.Headers
+import retrofit2.http.POST
+import retrofit2.http.Path
+import retrofit2.http.Query
+
 interface API {
-    @GET(value = "/api")
-    suspend fun getProduct() : ProductModel
+    @POST(value = "auth/local")
+    @Headers("No-Authentication: true")
+    suspend fun login(@Body loginModel: LoginModel ) : Response<LoginResponse>
+
+    @GET("products/{productId}?")
+    suspend fun getProduct(@Path("productId") productId: Int?, @Query("populate") populate: String?): Response<ProductModel>?
```

Comment:
While the populate part is indeed optional (tried it in postman), we won't be using the endpoints without it, so I think it's better to have it hardcoded here so repo/viewmodel methods don't need to specify it. I think there's no need to refactor this one right now as it is not incorrect, just a hint to not spend that much time and think about it for the other couple of endpoints. 

`productId`, on the other hand, definitely doesn't need to be optional. It doesn't make sense to get a product by ID without it.